src/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -930,33 +856,41 @@ public void BeginGetRequestStream_CreatePostRequestThenCallTwice_ThrowsInvalidOp
         {
             HttpWebRequest request = HttpWebRequest.CreateHttp(remoteServer);
             request.Method = "POST";
+            RequestState state = new RequestState();
+            state.Request = request;
 
-            IAsyncResult asyncResult = request.BeginGetRequestStream(null, null);
+            IAsyncResult asyncResult = request.BeginGetRequestStream(RequestCallback, state);
```

Comment:
This looks problematic: it's introducing a race condition, as the get request stream operation could complete before BeginGetRequestStream is called, at which point that should actually succeed rather than throw.